packages/nimble-components/src/table/index.ts

```diff
@@ -1006,15 +1071,11 @@ export class Table<
         });
     };
 
-    private toggleGroupExpanded(rowIndex: number): void {
-        const row = this.table.getRowModel().rows[rowIndex]!;
-        const wasExpanded = row.getIsExpanded();
-        // must update the collapsedRows before toggling expanded state
-        if (wasExpanded) {
-            this.collapsedRows.add(row.id);
-        } else {
-            this.collapsedRows.delete(row.id);
-        }
+    private toggleRowExpanded(rowIndex: number): void {
+        const rows = this.table.getRowModel().rows;
+        const row = rows[rowIndex]!;
+        // Must update the expansionManager before toggling expanded state
+        this.expansionManager.toggleRowExpansion(row);
         row.toggleExpanded();
```

Comment:
now we have two parallel systems tracking the expansion state of rows :/
Is that necessary? Can we just rely on the tanstack state? is there a path towards having just tanstack represent that state? (feature request? bug?) may be worth a discussion

Reply:
See [this comment thread](https://github.com/ni/nimble/pull/1663#discussion_r1456538472).